src/chunked.rs

```diff
@@ -215,20 +229,22 @@ impl<T> ChunkedAllocator<T> {
     ///
     /// ### Parameters:
     ///
-    /// - `blocks_per_chunk`: used for calculating size of memory blocks to request from the
-    ///                       underlying allocator
-    /// - `min_block_size`: ?
-    /// - `max_chunk_size`: ?
-    /// - `id`: hal memory type
+    /// - `blocks_per_chunk`: The number of blocks in each chunk allocated from the underlying
+    ///                       allocator.
+    /// - `min_block_size`: The minimum block size used by this allocator in bytes. Allocations
+    ///                     significantly than this may incur much larger overhead.
+    /// - `max_chunk_size`: The maximum size of chunks allocated from the underlying allocator
```

Comment:
You *sure*? Reading through the code, it is perfectly willing to allocate chunks with less than `blocks_per_chunk` blocks, when running up against `max_chunk_size`

Reply:
Uh. I forgot. Sorry 😏

Reply:
Actually, with that in mind, `CombinedAllocator` should probably allocate in `root` rather than `combined` when `reqs.size > self.chunks.max_chunk_size() / 2` rather than when `reqs.size > self.chunks.max_chunk_size()`.

(Since it'd take up the whole chunk then, anyways)